### PR TITLE
Feat/update linting tools

### PR DESCRIPTION
**Pull request by petter-a**
Updates eslint configuration to reflect common rules and support typescript.
Also a separate prettier config is introduced.

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/602c13c73b93a7537d767e421a43afaa/pull-request/304)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1pe1ex2 Uppdatera .eslintrc med nyare config (och ändra till .eslintrc.js)](https://app.clickup.com/t/1pe1ex2)

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    semi: true,
+    singleQuote: true,
+    tabWidth: 4,
```
Comment (dannil76) [inline .prettierrc.js:4]: What is the reason to change from 2 spaces to 4?
I vote for keeping it as is ie. 2 spaces

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                'linebreak-style': ['error', 'unix'],
+                'callback-return': ['error'],
+                'block-spacing': ['error'],
+                'comma-style': ['error', 'last'],
+                'eol-last': ['error', 'always'],
+                'lines-between-class-members': ['error', 'always'],
+                'max-len': [
+                    'error',
+                    {
+                        code: 120,
```
Comment (dannil76) [inline .eslintrc.js:76]: What is the reason for this value? 120 chars is way to long. My suggestion is to keep it between 70 - 80 chars

**Review by dannil76 — CHANGES_REQUESTED** [review]
Left some comments.

This branch should only deal with rules and settings for linting, etc. Believes that we developers should talk to each other about how we should handle the transition to applying these lint changes.

File: .prettierrc.js (hunk 1 above)
Comment (petter-a) [reply .prettierrc.js]: Guess it is a matter of preference. I think code is way more readable that way. Have reverted it to 2 though.

File: .eslintrc.js (hunk 2 above)
Comment (petter-a) [reply .eslintrc.js]: Again a preference thingy. Typescript prints better on a wider width. I have reverted to 80.

File: .prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  semi: true,
+  singleQuote: true,
+  tabWidth: 2,
+  printWidth: 80,
+};
```
Comment (MasterKenth) [inline .prettierrc.js:6]: What is the motivation for these values? Ideally in my opinion this should be left blank (`{}`) as the prettier defaults should be all we need.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        'no-useless-catch': ['error'],
+        'no-useless-return': ['error'],
+        'no-empty': ['error'],
+        'no-extra-boolean-cast': ['error'],
+        'no-floating-decimal': ['error'],
+        'no-warning-comments': ['error'],
+        'no-irregular-whitespace': ['error'],
+        'no-delete-var': ['error'],
+        'no-undef-init': ['error'],
+        'no-redeclare': ['error'],
```
Comment (MasterKenth) [inline .eslintrc.js:55]: Why are all these rules explicitly defined? The ideal eslint config in my opinion should be very sparse. Take a look at the [app config](https://github.com/helsingborg-stad/app-mitt-helsingborg/blob/develop/.eslintrc.js) for example.

**Review by MasterKenth — CHANGES_REQUESTED** [review]
It's not clear what the motivation for exactly these configurations come from.

I also don't think this PR should include formatting all files according to the new rules. That should ideally be in a separate PR or done on a file-by-file basis if that is too conflicting. Is formatting automatically applied?

**Pull request closed without merging**